backend/src/main/java/asegroup1/api/daos/landregistry/LandRegistryDaoImpl.java

```diff
@@ -61,4 +71,50 @@ public void delete(String id) {
                     return landRegistryData;
                 }).collect(Collectors.toList());
     }
+    
+	public int updateAveragePrice(HashMap<String, Long> averagePrices) throws IOException, UnirestException {
+		int updatedRecords = 0;
+		EntityManager em = getEntityManager();
+
+		for (Entry<String, Long> averagePrice : averagePrices.entrySet()) {
+			PostCodeCoordinates coordsToUpdate = em.find(PostCodeCoordinates.class, averagePrice.getKey());
+
+			if (!coordsToUpdate.getAverageprice().equals(averagePrice.getValue())) {
+
+				// update local values
+				em.getTransaction().begin();
+				coordsToUpdate.setAverageprice(averagePrice.getValue());
+				em.merge(coordsToUpdate);
+
+				// write update to database
+				em.getTransaction().commit();
+				updatedRecords++;
+			}
+		}
+
+		em.close();
+
+		return updatedRecords;
+    }
+
+	@SuppressWarnings("unchecked")
+	public HashMap<String, List<String>> getAllPostcodes(String regex, boolean restrictToUnset, int groupCharSize) {
```

Comment:
change to `getMatchingPostcodes` as most of the time all postcodes will not be retrieved